### DISCUSSION TITLE
NuxtLink - treat absolute URLs as internal for client-side navigation

### DISCUSSION THREAD

**Discussion by Phyllon** [nuxt/nuxt#16301] [Questions] [answered]
Hi there, 
I need to have absolute URLs (incl. origin) used in `NuxtLink` for SEO reasons but I want them to be treated internally, so they behave just like relative URLs, e.g. '/path/to/resource.html'. 
This is needed for client-side navigation, because absolute URLs trigger SSR. The `:external="false"` property of `NuxtLink` seems not to work like expected. My setup along with CSR works just fine when using relative URLs.
Is there any way to solve this?

Thanks in advance!

Meta: 
Nuxt 3.0.0-rc.13

**Comment by TheAlexLichter** [comment, marked as answer]
Yes, this is indeed not supported and a very unusual request 🙈

Could you kindly elaborate the "SEO reasons" behind it? 🤔

**Reply by Phyllon** [reply]
Thank you for your reply! 
Our SEO team needs absolute URLs in the HTML of the page for the google bot when being crawled. That's all I know, sorry.
So there is no chance to solve this, e.g. by overriding anything in /app/router.options.ts for example?

**Reply by TheAlexLichter** [reply]
Well, you probably could create your own component based on `<NuxtLink>` and create such a behavior.

Besides that, the request regarding SEO seems to be quite weird. Instead of using absolute URLs, one should set canonicals correctly and configure the proxy / routing to use "one version" of the site.

**Reply by Phyllon** [reply]
After a lot of reading and adaption I could figure it out and posted the general answer here: 
https://stackoverflow.com/a/74561723/19901666

For this special problem the solution is:
```
<NuxtLink
  :to="'https://host.com/any/path'"
  @click.prevent="navigateTo('/any/path')"
>client-side nav</NuxtLink>
```

 
Thank you anyway <3
